app/controllers/projects/members_controller.rb

```diff
@@ -110,25 +109,51 @@ def find_member(str)
       User.find_by(login: str.to_s.sub(/ <.*>$/, ""))
     end
 
-    # Redirects back to show_project.
-    def update_member_status(project, candidate)
-      admin = member = :remove
-      case params[:commit]
-      when :change_member_status_make_admin.l
-        unless project.is_admin?(@user)
+    def update_membership(project, candidate)
+      unless update_trust_status(project, candidate)
+        unless update_admin_status(project, candidate)
           return must_be_project_admin!(project.id)
         end
 
-        admin = member = :add
-      when :change_member_status_make_member.l
-        member = :add
+        update_member_status(project, candidate)
+      end
+      return_to_caller(project, params[:target])
+    end
+
+    def update_trust_status(project, candidate)
+      if params[:commit] == :change_member_status_revoke_trust.l
+        flash_notice(:change_member_status_revoke_trust_flash.l)
+        set_trust(project, candidate, false)
+      elsif params[:commit] == :change_member_status_trust.l
+        flash_notice(:change_member_status_trust_flash.l)
+        set_trust(project, candidate, true)
+      else
+        return false
       end
+      true
```

Comment:
This may be idiosyncratic, but I'd find it easier to read if the return values were included inside each condition. (But that adds a line and a wee bit of repetition.) YMMV
Suggestion:
```ruby
        set_trust(project, candidate, false)
        true
      elsif params[:commit] == :change_member_status_trust.l
        flash_notice(:change_member_status_trust_flash.l)
        set_trust(project, candidate, true)
        true
      else
        false
      end
```

Reply:
Sure.  Rubocop is cool with so that's fine.